### PR TITLE
Breakout core runtime files

### PR DESCRIPTION
**Pull request by jwerle**
This PR breaks out core runtime concerns into their own files, moving inline implementations out of the headers, and subsequently improving the build time feedback loop with less changes occurring in the head files to trigger a complete rebuild. This PR renames various functions/macros for improved naming. It also removes the need for `common.hh` while also improving the header include surface for `init.cc`, meaning less downstream dependency requirements.

### CODE REVIEW

File: src/cli/cli.cc [hunk 1]
```diff
@@ -4751,7 +4797,7 @@ int main (const int argc, const char* argv[]) {
 
       // Check for gradle in pwd. Don't fail, this is just for support.
       if (!fs::exists(String("gradlew") + (platform.win ? ".bat" : ""))) {
-        log("WARN: gradlew script not in pwd: " + fs::current_path().string());
+        log("WARNING: gradlew script not in pwd: " + fs::current_path().string());
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:4800]: we can introduce `warn` and `error` functions (later) to prefix automatically

Comment (jwerle) [reply src/cli/cli.cc:4800]: 👍🏼

**Review by bcomnes — APPROVED** [review]
Big organizational improvement.

File: api/notification.js [hunk 2]
```diff
@@ -876,7 +876,7 @@ export class Notification extends EventTarget {
 }
 
 hooks.onReady(() => {
-  if (os.host() === 'iphone-simulator' || os.host() === 'android-emulator') {
+  if (os.host() === 'iphonesimulator' || os.host() === 'android-emulator') {
```
[the remark this answers is not in the document]
Comment (jwerle) [reply api/notification.js:879]: it is the same value used with `xcrun -sdk` - we could include a `-` if you think appropriate

**Pull request merged**